kettle-plugins/hdfs/src/main/java/org/pentaho/big/data/kettle/plugins/hdfs/trans/HadoopFileInputMeta.java

```diff
@@ -210,4 +216,26 @@ FileInputList createFileList( VariableSpace space ) {
     return FileInputList.createFileList( space, inputFiles.fileName, inputFiles.fileMask, inputFiles.excludeFileMask,
       inputFiles.fileRequired, inputFiles.includeSubFolderBoolean() );
   }
+
+  private String encryptDecryptPassword( String source, EncryptDirection direction ) {
+    try {
+      URI uri = new URI( source );
+      String userInfo = uri.getUserInfo();
+      if ( userInfo != null ) {
+        String[] userInfoArray = userInfo.split( ":", 2 );
+        String password = userInfoArray[ 1 ];
+        String processedPassword = password;
+        if ( direction == EncryptDirection.ENCRYPT ) {
```

Comment:
We should replace this by a switch w/ a default fallthrough throwing an invalidparameterexception or something like that.